shlr/bochs/src/libbochs.c

```diff
@@ -217,6 +247,79 @@ bool bochs_open(libbochs_t* b, const char * rutaBochs, const char * rutaConfig)
 			}
 		}
 	}
+#else
+	#define PIPE_READ 0
+	#define PIPE_WRITE 1
+	int aStdinPipe[2];
+	int aStdoutPipe[2];
+	int nChild;
+	int nResult;
+	char *newargv[] = { rutaBochs, "-q","-f", rutaConfig, NULL };
+        char *envi[] = { "DISPLAY=:0.0", NULL };
+	if (pipe (aStdinPipe) < 0) {
+		eprintf ("Error: allocating pipe for child input redirect");
+		return false;
+	}
+	if (pipe(aStdoutPipe) < 0) {
+		close (aStdinPipe[PIPE_READ]);
+		close (aStdinPipe[PIPE_WRITE]);
+		eprintf ("Error: allocating pipe for child output redirect");
+		return false;
+	}
+
+	nChild = fork();
+	if (0 == nChild) {
+		// redirect stdin
+		if (dup2 (aStdinPipe[PIPE_READ], STDIN_FILENO) == -1) {
+			eprintf("Error: redirecting stdin");
+			return false;
+		}
+
+		// redirect stdout
+		if (dup2 (aStdoutPipe[PIPE_WRITE], STDOUT_FILENO) == -1) {
+			eprintf("Error: redirecting stdout");
+			return false;
+		}
+
+		// redirect stderr
+		if (dup2 (aStdoutPipe[PIPE_WRITE], STDERR_FILENO) == -1) {
+			eprintf("Error: redirecting stderr");
+			return false;
+		}
+
+		close (aStdinPipe[PIPE_READ]);
+		close (aStdinPipe[PIPE_WRITE]);
+		close (aStdoutPipe[PIPE_READ]);
+		close (aStdoutPipe[PIPE_WRITE]);
+		//eprintf("Execv %s\n",rutaBochs);
+		nResult = execve(rutaBochs, newargv, envi);
```

Comment:
execve replaces current process with a new one. so it never returns unless it fails hard. so you must perror + exit() here to avoid having 2 radares running